cpp/dominatingset/BranchAndReduceMSC.cpp

```diff
@@ -0,0 +1,87 @@
+#include <dominatingset/RooijBodlaenderMDS.hpp>
+#include <dominatingset/BranchAndReduceMSC.hpp>
+#include <cassert>
+
+NetworKit::index findCountingRuleReductionSet(std::vector<std::set<NetworKit::node>> &family, std::vector<std::set<NetworKit::index>> &occurences);
```

Comment:
Please run the following command and fix linter errors:

cpplint --linelength=100 --extensions=cpp,hpp --filter=-legal/copyright,-build/c++20,-build/include,-build/namespaces,-build/header_guard,-runtime/references,-runtime/string --recursive --exclude=./backup/* --exclude=./build/* --exclude=./input/* --exclude=./lib/* --exclude=./cpp/flow/maximum_flow/dynamic_tree/* .